gax/src/main/java/com/google/api/gax/rpc/CancelledException.java

```diff
@@ -39,4 +39,9 @@ public CancelledException(
       String message, Throwable cause, StatusCode statusCode, boolean retryable) {
     super(message, cause, statusCode, retryable);
   }
+
+  public CancelledException(
+      Throwable cause, StatusCode statusCode, boolean retryable, ErrorDetails errorDetails) {
+    super(cause, statusCode, retryable, errorDetails);
```

Comment:
Subjective: For this an all similar changes, can you make them such that each of the overloaded constructors call `this()` (except the last one, which does the job). This is an idiomatic way of doing it and is often easier to maintain

Reply:
Do you mean making the first constructor call `this(cause, statusCode, retryable, null)`?

Reply:
yep. It is a typical approach: to have some sort of a "master" consturctor and make other constructors simply call the big one. The master constructor can even be private, if we don't want to expose it.

Reply:
I agree that this is ideal if possible, but I think it depends on whether `super(field1, field2)` has the same behavior as `this(field1, field2, null)`, sometimes they don't. For example, `super(cause)` and `this(null, cause)` in `RuntimeException` have different behaviors, one will populate message from cause, the other will set it to null. That being said, since this is a side refactoring, maybe we can merge the main logic of the code first and come back to see if we can improve it.